### PR TITLE
CRUD termination

### PR DESCRIPTION
**Pull request by jdml-ml**
## What?
I have added the "C" of CRUD, this allows me to create a new article following the architecture of MVC. Also, the form builder was used to configure following the Rails conventions. Some validations and error messages were added for the correct handling of exceptions that the user can generate through his inputs.
Another aspect added was the "U" to allow updating an existing item in the list. This also makes use of exception handling to handle errors.
Factored the view into a "_form.html.erb" file to reuse code in the view.
Finally, the "D (Delete)" function was added making use of Resourceful Routing

## Why?
These changes complete the CRUD functionality of the program

## How?
Important CRUD concepts were included. Also, code reuse through shared views

## Anything Else?
After removing an article, for example with id=1, if another one is added again, it will already be id=2. I want to delve into why it was not saved with id=1 since the list of articles was already empty

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -2,7 +2,49 @@ class ArticlesController < ApplicationController
...
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @article = Article.find(params[:id])
+    @article.destroy
+
+    redirect_to root_path, status: :see_other
```
Comment (estebancordoba) [inline app/controllers/articles_controller.rb:42]: Where does this line redirect us?

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -2,7 +2,49 @@ class ArticlesController < ApplicationController
...
+  end
+
+  private
+    def article_params
+      params.require(:article).permit(:title, :body)
```
Comment (estebancordoba) [inline app/controllers/articles_controller.rb:47]: What does permit do?

File: app/models/article.rb [hunk 3]
```diff
@@ -1,2 +1,5 @@
 class Article < ApplicationRecord
-end
+    validates :title, presence: true
+    validates :body, presence: true, length: { minimum: 10 }
```
Comment (estebancordoba) [inline app/models/article.rb:3]: What does presence do?

Comment (jdml-ml) [reply app/models/article.rb:3]: It serves to validate that this attribute is not empty or `nil`

File: app/controllers/articles_controller.rb (hunk 2 above)
Comment (jdml-ml) [reply app/controllers/articles_controller.rb:47]: Serves to provide security to the system, if not specified what parameters it have permission from the front could inject attributes of the class. Therefore, the function sets which parameters are allowed to be modified by the user

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (jdml-ml) [reply app/controllers/articles_controller.rb:42]: This line redirects to the `root` route that we have defined as the `root` (this route you can find in file routes.rb).

**Review by jdml-ml — COMMENTED** [review]
Sent comments

**Review by estebancordoba — APPROVED** [review]

**Pull request merged**